holoviews/operation/datashader.py

```diff
@@ -1520,20 +1520,31 @@ def _apply_spreading(self, array):
         """Apply the spread function using the indicated parameters."""
         raise NotImplementedError
 
-    def _process(self, element, key=None):
-        if not isinstance(element, RGB):
-            raise ValueError('spreading can only be applied to RGB Elements.')
-        rgb = element.rgb
-        new_data = {kd.name: rgb.dimension_values(kd, expanded=False)
-                    for kd in rgb.kdims}
+    def _preprocess_rgb(self, element):
         rgbarray = np.dstack([element.dimension_values(vd, flat=False)
                               for vd in element.vdims])
-        data = self.uint8_to_uint32(rgbarray)
+        if rgbarray.dtype.kind == 'f':
+            rgbarray = rgbarray * 255
+        return tf.Image(self.uint8_to_uint32(rgbarray.astype('uint8')))
+
+    def _process(self, element, key=None):
+        if isinstance(element, RGB):
+            rgb = element.rgb
+            data = self._preprocess_rgb(rgb)
+        elif isinstance(element, Image):
+            data = element.clone(datatype=['xarray']).data[element.vdims[0].name]
+        else:
+            raise ValueError('spreading can only be applied to Image or RGB Elements.')
```

Comment:
What about spreading on categorical rasterized data (which I believe is an NdOverlay of Image)? Is that already handled separately so that the operation is applied per Image?